exchanges/bitfinex/bitfinex_websocket.go

```diff
@@ -182,897 +184,929 @@ func (b *Bitfinex) wsHandleData(respRaw []byte) error {
-					}
-					period, ok := data[4].(float64)
-					if !ok {
-						return errors.New("unable to type assert trade period")
-					}
-					wsTrade.Rate = rate
-					wsTrade.Period = int64(period)
-				} else {
-					price, ok := data[3].(float64)
-					if !ok {
-						return errors.New("unable to type assert trade price")
-					}
-					wsTrade.Price = price
+				wsFundingTrade.MTSCreated = time.UnixMilli(int64(created))
+				offerID, ok := data[3].(float64)
+				if !ok {
+					return errors.New("unable to type assert funding trade offer ID")
 				}
-				tradeHolder = append(tradeHolder, wsTrade)
-			}
-			trades := make([]trade.Data, len(tradeHolder))
-			for i := range tradeHolder {
-				side := order.Buy
-				newAmount := tradeHolder[i].Amount
-				if newAmount < 0 {
-					side = order.Sell
-					newAmount *= -1
+				wsFundingTrade.OfferID = int64(offerID)
+				if wsFundingTrade.Amount, ok = data[4].(float64); !ok {
+					return errors.New("unable to type assert funding trade amount")
 				}
-				price := tradeHolder[i].Price
-				if price == 0 && tradeHolder[i].Rate > 0 {
-					price = tradeHolder[i].Rate
+				if wsFundingTrade.Rate, ok = data[5].(float64); !ok {
+					return errors.New("unable to type assert funding trade rate")
 				}
-				trades[i] = trade.Data{
-					TID:          strconv.FormatInt(tradeHolder[i].ID, 10),
-					CurrencyPair: pair,
-					Timestamp:    time.UnixMilli(tradeHolder[i].Timestamp),
-					Price:        price,
-					Amount:       newAmount,
-					Exchange:     b.Name,
-					AssetType:    chanAsset,
-					Side:         side,
+				period, ok := data[6].(float64)
+				if !ok {
+					return errors.New("unable to type assert funding trade period")
 				}
+				wsFundingTrade.Period = int64(period)
+				wsFundingTrade.Maker = data[7] != nil
+				b.Websocket.DataHandler <- wsFundingTrade
+			}
+		default:
+			b.Websocket.DataHandler <- stream.UnhandledMessageWarning{
+				Message: b.Name + stream.UnhandledMessage + string(respRaw),
 			}
+			return nil
+		}
+	}
+	return nil
+}
+
+func (b *Bitfinex) handleWSChannelUpdate(c *stream.ChannelSubscription, chanID int, eventType string, d []interface{}) error {
+	if eventType == wsChecksum {
+		return b.handleWSChecksum(chanID, d)
+	}
+
+	if eventType == wsHeartbeat {
+		return nil
+	}
+
+	switch c.Channel {
+	case wsBook:
+		return b.handleWSBookUpdate(c, chanID, d)
+	case wsCandles:
+		return b.handleWSCandleUpdate(c, d)
+	case wsTicker:
+		return b.handleWSTickerUpdate(c, d)
+	case wsTrades:
+		return b.handleWSTradesUpdate(c, eventType, d)
+	}
+
+	return fmt.Errorf("%s unhandled channel update: %s", b.Name, c.Channel)
+}
+
+func (b *Bitfinex) handleWSChecksum(chanID int, d []interface{}) error {
+	var token int
+	if f, ok := d[2].(float64); !ok {
+		return common.GetTypeAssertError("float64", d[2], "checksum")
+	} else {
+		token = int(f)
+	}
+
+	var seqNo int64
+	if f, ok := d[3].(float64); !ok {
+		return common.GetTypeAssertError("float64", d[3], "seqNo")
+	} else {
```

Comment:
I think you're going to need to `//nolint:indent-error-flow // using temporary variable requires else statement` (or nolint:revive) otherwise this will likely make all our PR linting fail

Reply:
Or could even do this:

```go
seqNo , ok := d[3].(float64)
if !ok {
	return common.GetTypeAssertError("float64", d[3], "seqNo")
}
```
<img width="218" alt="image" src="https://github.com/thrasher-corp/gocryptotrader/assets/9391715/6e176d0d-5bd3-401c-a5c4-1e35d365dbd5">